### DISCUSSION TITLE
Support Vite

### DISCUSSION THREAD

**Discussion by rtritto** [uNetworking/uWebSockets.js#1120] [Ideas]
https://github.com/vitejs/vite/issues/16579#issuecomment-2428060674

> I checked uWebSockets.js and I think it has the following problems to be used in Vite:
> 
>   - lack of minor platform support: it only supports [Tier 1 platforms](https://github.com/nodejs/node/blob/main/BUILDING.md#platform-list) and there aren't any fallback implementations like we have with esbuild
> 
>   - does not have a builtin way to only install the binary needed for the platform it is ran in (e.g. `optionalDependencies`)
> 
> 
> Even if these problems were solved, it has a big package size even for a single platform (5MB; ws is 143kB), so it would need to improve the perf significantly to make it worth.

**Comment by uNetworkingAB** [comment]
I have no idea what vite is. But if they have a problem with software taking up 4 floppy disks of space, sounds like they aren't target audience

**Reply by rtritto** [reply]
Vite is a popular web development build tool designed for development of front-end and server.

Second point can be achieved with #1112.

What about the first point? (`esbuild` is a bundler)
